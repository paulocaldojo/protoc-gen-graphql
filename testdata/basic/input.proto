syntax = "proto3";

package protoc_gen_graphql.test.basic;

option go_package = "github.com/paulocaldojo/protoc-gen-graphql/testdata/basic";

message ScalarFields {
  string string_field = 1;
  bytes bytes_field = 2;
  int32 int32_field = 3;
  uint32 uint32_field = 4;
  sint32 sint32_field = 5;
  fixed32 fixed32_field = 6;
  int64 int64_field = 7;
  uint64 uint64_field = 8;
  sint64 sint64_field = 9;
  fixed64 fixed64_field = 10;
  float float_field = 11;
  double double_field = 12;
  bool bool_field = 13;
  string deprecated_field = 14 [deprecated = true];
}

// This is a comment attached to the EnumType message.
// This is its second line.
//
// And this is its second paragraph.
enum EnumType {
  // This is a leading comment attached to the UNKNOWN value.
  // This is its second line.
  //
  // And this is its second paragraph.
  UNKNOWN = 0;
  FIRST = 1; // This is a comment attached to the FIRST value.
  SECOND = 2;
  DEPRECATED = 3 [deprecated = true];
}

enum UnusedEnum {
  ZERO = 0;
}

message ComplexFields {
  ScalarFields message_field = 1;
  EnumType enum_field = 2;
  repeated string repeated_scalar_field = 3;
  repeated ScalarFields repeated_message_field = 4;
  map<string, EnumType> map_field = 5;
  oneof oneof_field {
    string first = 6;
    ScalarFields second = 7;
    string deprecated = 8 [deprecated = true];
  }
  Empty empty_field = 9;
}

message Outer {
  message Middle {
    message Inner {
      string inner_name = 1;
    }
    string middle_name = 1;
  }
  Middle middle = 1;
  Middle.Inner inner = 2;
}

message Empty {}

// This is a detached comment for Comment.

// This is a comment attached to the Comment message.
// This is its second line.
//
// And this is its second paragraph.
message Comment {
  // This is a comment attached to the NestedComment message.
  // This is its second line.
  //
  // And this is its second paragraph.
  message NestedComment {
    // This is a leading comment attached to the nested_foo field.
    // This is its second line.
    //
    // And this is its second paragraph.
    string nested_foo = 1;
  }

  // This is a leading comment attached to the NestedEnum enum.
  // This is its second line.
  //
  // And this is its second paragraph.
  enum NestedEnum {
    // This is a leading comment attached to the ZERO value.
    // This is its second line.
    //
    // And this is its second paragraph.
    ZERO = 0;
  }

  // This is a leading comment attached to the foo field.
  // This is its second line.
  //   - This is its third line.
  //
  // And this is its second paragraph.
  string foo = 1;

  // This is a leading comment attached to the bar field.
  // This is its second line.
  //
  // And this is its second paragraph.
  string bar = 2;
  // This is a trailing comment attached to the bar field.
  // This is its second line.
  //
  // And this is its second paragraph.

  /*
  This is an a block comment attached to the baz field.
  This is its second line.
    - This is its third line.

  And this is its second paragraph.
  */
  string baz = 3;

  // This is a detached comment for qux.

  /* This is a one line comment attached to the qux field. */
  string qux = 4;

  // This is a leading comment attached to the oneof.
  oneof oneof {
    // This is a leading comment attached to the first oneof value.
    string first = 5;
  }
}
