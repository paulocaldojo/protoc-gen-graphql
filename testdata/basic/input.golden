# DO NOT EDIT! Generated by protoc-gen-graphql.

type ProtocGenGraphqlTestBasic_ScalarFields {
  stringField: String!
  bytesField: String!
  int32Field: Int!
  uint32Field: Int!
  sint32Field: Int!
  fixed32Field: Int!
  int64Field: Int!
  uint64Field: Int!
  sint64Field: Int!
  fixed64Field: Int!
  floatField: Float!
  doubleField: Float!
  boolField: Boolean!
  deprecatedField: String! @deprecated
}

type ProtocGenGraphqlTestBasic_ComplexFields {
  messageField: ProtocGenGraphqlTestBasic_ScalarFields
  enumField: ProtocGenGraphqlTestBasic_EnumType!
  repeatedScalarField: [String!]!
  repeatedMessageField: [ProtocGenGraphqlTestBasic_ScalarFields!]!
  mapField: [ProtocGenGraphqlTestBasic_ComplexFields_MapFieldEntry!]!
  oneofField: ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof
  emptyField: ProtocGenGraphqlTestBasic_Empty
}

"""
`ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof` represents the `oneof_field` oneof in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
union ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof = ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_First | ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Second | ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Deprecated

"""
`ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_First` represents the `first` oneof field in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
type ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_First {
  _typename: String
  first: String!
}

"""
`ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Second` represents the `second` oneof field in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
type ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Second {
  _typename: String
  second: ProtocGenGraphqlTestBasic_ScalarFields
}

"""
`ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Deprecated` represents the `deprecated` oneof field in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
type ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Deprecated {
  _typename: String
  deprecated: String! @deprecated
}

"""
`ProtocGenGraphqlTestBasic_ComplexFields_MapFieldEntry` represents the `map_field` map in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
type ProtocGenGraphqlTestBasic_ComplexFields_MapFieldEntry {
  key: String!
  value: ProtocGenGraphqlTestBasic_EnumType!
}

type ProtocGenGraphqlTestBasic_Outer {
  middle: ProtocGenGraphqlTestBasic_Outer_Middle
  inner: ProtocGenGraphqlTestBasic_Outer_Middle_Inner
}

type ProtocGenGraphqlTestBasic_Outer_Middle {
  middleName: String!
}

type ProtocGenGraphqlTestBasic_Outer_Middle_Inner {
  innerName: String!
}

type ProtocGenGraphqlTestBasic_Empty {
  _empty: Boolean
}

"""
This is a comment attached to the Comment message.
This is its second line.

And this is its second paragraph.
"""
type ProtocGenGraphqlTestBasic_Comment {
  """
  This is a leading comment attached to the foo field.
  This is its second line.
    - This is its third line.

  And this is its second paragraph.
  """
  foo: String!
  """
  This is a leading comment attached to the bar field.
  This is its second line.

  And this is its second paragraph.

  This is a trailing comment attached to the bar field.
  This is its second line.

  And this is its second paragraph.
  """
  bar: String!
  """
  This is an a block comment attached to the baz field.
  This is its second line.
  - This is its third line.

  And this is its second paragraph.
  """
  baz: String!
  """
  This is a one line comment attached to the qux field.
  """
  qux: String!
  """
  This is a leading comment attached to the oneof.
  """
  oneof: ProtocGenGraphqlTestBasic_Comment_OneofOneof
}

"""
`ProtocGenGraphqlTestBasic_Comment_OneofOneof` represents the `oneof` oneof in `protoc_gen_graphql.test.basic.Comment`.
"""
union ProtocGenGraphqlTestBasic_Comment_OneofOneof = ProtocGenGraphqlTestBasic_Comment_OneofOneof_First

"""
`ProtocGenGraphqlTestBasic_Comment_OneofOneof_First` represents the `first` oneof field in `protoc_gen_graphql.test.basic.Comment`.
"""
type ProtocGenGraphqlTestBasic_Comment_OneofOneof_First {
  _typename: String
  """
  This is a leading comment attached to the first oneof value.
  """
  first: String!
}

"""
This is a comment attached to the NestedComment message.
This is its second line.

And this is its second paragraph.
"""
type ProtocGenGraphqlTestBasic_Comment_NestedComment {
  """
  This is a leading comment attached to the nested_foo field.
  This is its second line.

  And this is its second paragraph.
  """
  nestedFoo: String!
}

"""
This is a leading comment attached to the NestedEnum enum.
This is its second line.

And this is its second paragraph.
"""
enum ProtocGenGraphqlTestBasic_Comment_NestedEnum {
  """
  This is a leading comment attached to the ZERO value.
  This is its second line.

  And this is its second paragraph.
  """
  ZERO
}

"""
This is a comment attached to the EnumType message.
This is its second line.

And this is its second paragraph.
"""
enum ProtocGenGraphqlTestBasic_EnumType {
  """
  This is a leading comment attached to the UNKNOWN value.
  This is its second line.

  And this is its second paragraph.
  """
  UNKNOWN
  """
  This is a comment attached to the FIRST value.
  """
  FIRST
  SECOND
  DEPRECATED @deprecated
}

enum ProtocGenGraphqlTestBasic_UnusedEnum {
  ZERO
}
